app/components/formatted_time.js

```diff
@@ -28,26 +29,20 @@ export default class FormattedTime extends React.PureComponent {
         } = this.props;
 
         if (timeZone) {
-            return intl.formatDate(value, {
-                timeZone,
+            moment.tz.setDefault(timeZone);
+            return intl.formatDate(moment(value).toDate(), {
```

Comment:
**moment.tz.setDefault** will change timezone in all moment calls. 
Use 
`return intl.formatDate(moment(value, timeZone).toDate())`

script to check - https://gist.github.com/sergeyzhukov/0d2f552df7a9cbc64833df849d083ffc

Reply:
So, in that case, maybe it's better to do moment.tz.setDefault(timeZone) only one time in some global place? I saw you doing moment.tz.setDefault(timeZone) in two different components.

Reply:
@sergeyzhukov I went with your first suggestion as I could not find a good place to set the default just once